modules/tinymce/src/core/main/ts/content/TransparentElements.ts

```diff
@@ -0,0 +1,57 @@
+import { Arr, Obj, Type } from '@ephox/katamari';
+import { Compare, SugarElement, SugarNode, Traverse } from '@ephox/sugar';
+
+import AstNode from '../api/html/Node';
+import Schema, { SchemaMap } from '../api/html/Schema';
+import * as NodeType from '../dom/NodeType';
+
+export const transparentBlockAttr = 'data-mce-block';
+
+const makeSelectorFromSchemaMap = (map: SchemaMap) => Arr.filter(Obj.keys(map), (key) => /^[a-z]+$/.test(key)).join(',');
+
+const updateAt = (schema: Schema, root: Element, scope: Element, inEditorRoot: boolean) => {
+  const transparentSelector = makeSelectorFromSchemaMap(schema.getTransparentElements());
+  const blocksSelector = makeSelectorFromSchemaMap(schema.getBlockElements());
```

Comment:
As this is only used if a transparent element is found I wonder if we should be lazily doing this (e.g `Thunk.cached`)? I'm not sure on the performance impact though, but this doe happen on every parse 🤷

Reply:
Not sure Thunk.cached can be used here since it would create a new thunk each time. It would have to be some global state bag. Maybe we optimize this later if it need to be optimized. The sets here are pretty small.